File: src/main/java/com/core/data_pipeline_platform/domain/file/dto/ChunkUploadRequest.java
package com.core.data_pipeline_platform.domain.file.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PositiveOrZero;

public record ChunkUploadRequest(
        @NotBlank String sessionId,
        @PositiveOrZero int chunkIndex,
        @NotNull byte[] chunkData,
        @PositiveOrZero long chunkSize
) {}

File: src/main/java/com/core/data_pipeline_platform/domain/file/dto/ChunkUploadRequest.kt
package com.core.data_pipeline_platform.domain.file.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.PositiveOrZero

data class ChunkUploadRequest(
    val sessionId: @NotBlank String?,
    val chunkIndex: @PositiveOrZero Int,
    val chunkData: ByteArray?,
    val chunkSize: @PositiveOrZero Long
)
